### PR TITLE
Added schemas and mappers using pydantic, added database support

### PR DESCRIPTION
**Pull request by veyisTurgut**
Added base model schemas and corresponding mappers, also updated [get] /books endpoint.

Documentation of endpoint can be found here: https://github.com/bounswe/2021SpringGroup12/wiki/Practice-app-Documentation#1-get-books

### CODE REVIEW

File: practice-app/README.md [hunk 1]
```diff
@@ -24,4 +24,4 @@ Here we have an API, written using The Flask framework.
 * Go to the browser and open that link just you seen in the terminal.  (likely http://127.0.0.1:5000/)
 * Clone the repo to your local and branch out. 
 * Write your own endpoints using different existing API's.
-* Test your endpoints using Postman. Try to use query-path parameters, request bodies, etc. 
+* Test you endpoint using Postman. Try to use query-path parameters, request bodies, etc.
```
Comment (yamanogluberk) [inline practice-app/README.md:27]: Typo?

Comment (veyisTurgut) [reply practice-app/README.md:27]: probably, other than that?

File: practice-app/schemas.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from typing import List, Optional
+from pydantic import BaseModel
+
+
+class CreateBook(BaseModel):
```
Comment (yamanogluberk) [inline practice-app/schemas.py:5]: It looks like the CreateBook class is not used and it is very similar to the Book class, can you explain why do we need it?

**Review by yamanogluberk — COMMENTED** [review]
I had a few questions and attached them as comments.

File: practice-app/schemas.py (hunk 2 above)
Comment (veyisTurgut) [reply practice-app/schemas.py:5]: I will need this in post method. Users will enter these fields to store a book in database.

Comment (yamanogluberk) [reply practice-app/schemas.py:5]: I see, it makes sense.

**Comment by veyisTurgut** [comment]
now you can easily connect to our database and use it. Also added post method of /books endpoint. Unittest and documentations coming soon.

File: practice-app/practice-app.py [hunk 3]
```diff
@@ -3,58 +3,102 @@
...
+    if "name" not in request.args:
+        return Response("Please provide an author name!", status=400)
+    # now we are sure name parameter is supplied, request books of this author from NYTimes API.
     author_name = request.args.get("name").title().replace(" ","+")
     r = requests.get("https://api.nytimes.com/svc/books/v3/reviews.json?author={}&api-key=Ug3r4KRJxcd69bNw5i89CWxArqiGlrGB".format(author_name))
     r = r.json()
+    # now we have books of this author in r.
+    books = [mapper.book_mapper(s) for s in r["results"]]
+    # Let's store this book in database for further references.
+    con = sqlite3.connect("./sqlfiles/practice-app.db")
```
[the remark this answers is not in the document]
Comment (veyisTurgut) [reply practice-app/practice-app.py:41]: That's a great idea, we can send SQL code and parameters to a function and get the data as return value.
Let me implement it until tomorrow.

**Pull request merged**